taraxa/state/state_transition/state_transition.go

```diff
@@ -106,7 +104,7 @@ func (st *StateTransition) BlockNumber() types.BlockNum {
 func (st *StateTransition) BeginBlock(blk_info *vm.BlockInfo) {
 	st.begin_block()
 	blk_n := st.pending_blk_state.GetNumber()
-	rules_changed := st.evm.SetBlock(&vm.Block{Number: blk_n, BlockInfo: *blk_info} /*st.chain_config.EVMChainConfig.Rules(blk_n)*/)
+	rules_changed := st.evm.SetBlock(&vm.Block{Number: blk_n, BlockInfo: *blk_info}, st.chain_config.Hardforks.Rules(blk_n))
 	if st.dpos_contract != nil && rules_changed {
 		st.dpos_contract.Register(st.evm.RegisterPrecompiledContract)
 	}
```

Comment:
This rules changed will trigger registration of DPOS contract again. Is it ok, won't it break something?